sdk/storage/src/core/perform_request_response.rs

```diff
@@ -0,0 +1,23 @@
+use hyper::client::ResponseFuture;
+use url::Url;
+
+#[derive(Debug)]
+pub struct PerformRequestResponse {
+    pub(crate) url: Url,
+    pub(crate) response_future: ResponseFuture,
```

Comment:
Does it make sense to impl Future on `PerformRequestResponse` so that you can just call await on it?

Reply:
I believe making `check_status_extract_headers_and_body` a member function we achieved the same thing.